File: 1-two-sum/1-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        result = {}
        
        for i in range(len(nums)):
            if target - nums[i] in result:
                return [i, result[target - nums[i]]]
            
            result[nums[i]] = i

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> enumeration = new HashMap<>();
        int[] result = new int[2];
        
        for (int i = 0; i < nums.length; i++) {
            if (enumeration.containsKey(target - nums[i])) {
                result[0] = i;
                result[1] = enumeration.get(target - nums[i]);
            }
            
            enumeration.put(nums[i], i);
        }
        
        return result;
    }
}